react/react1/week3/react1week3-shifts/src/components/InputShift.js

```diff
@@ -0,0 +1,98 @@
+import React, { useState } from 'react';
+import { ViewShiftList } from './ViewShiftList';
+import { SearchShift } from './SearchShift';
+
+import { DataDefaultShiftApi } from './DataDefaultShiftApi';
+
+export function InputShift() {
+  const [inputShift, setInputShift] = useState([]);
+
+  const [name, setName] = useState('');
+  const [start, setStartTime] = useState('');
+  const [end, setEndTime] = useState('');
+  const [searchName, setSearchName] = useState('');
+  const [dataFechingState, setDataFechingState] = useState(false);
+
+  function addShift() {
+    const startEntry = start.split(':');
+    const startMinutes = Number(startEntry[0]) * 60 + Number(startEntry[1]);
+    const endEntry = end.split(':');
+    const endMinutes = Number(endEntry[0]) * 60 + Number(endEntry[1]);
+    const totalMinutes = endMinutes - startMinutes;
+    const hours = Math.floor(totalMinutes / 60);
+    const minutes = totalMinutes % 60;
+
+    if (hours > 0 || minutes > 0) {
+      const addedShift = [...inputShift, { name, start, end }];
+      setInputShift(addedShift);
+      alert('shift saved');
+    } else {
+      setInputShift(inputShift);
+      alert('check the time');
+    }
+  }
+
+  function deleteShift(name) {
+    setInputShift((shifts) => {
+      const shiftsAfterDelete = shifts.filter((shift) => shift.name !== name);
```

Comment:
Normally in projects every item should have a id number. You can use that id for key and filter. In this case if you want to remove an shift if same person joined different shifts then his all shifts will deleted from state. Because this will delete all same names from state. Maybe it could be different for change this project for now but you can be more carefull next projects.

Reply:
i am aware of it. i hade done in that way but we do not have id in api so changed it to this way

Reply:
i fixed everyghin except the delete method with name instead of  id.  in real project it wont be like that. it should be deleted by id.  i am aware of it in this project. can you check finally if everything is ok. so i will merge it.

Reply:
i also fixed merge conflict !

Reply:
now i fixed everything. also delete method.